src/linear_samplers/sampler_helpers/hadamard.jl

```diff
@@ -0,0 +1,46 @@
+"""
+fwht!(x::AbstractVector; signs=ones(Bool, size(x)), scaling = 1)  
+
+Performs an in-place Fast Walsh Hadamard Transform (FWHT). `signs` allows the user to input a booleanvector that can flip the signs in the transform. `scaling` allows the user to scale the result of the transform. 
+
+!!! Note: To avoid log computation at every call the function does not check that the dimension is a power of 2. This must be done by a separate function at an earlier point.
+"""
+function fwht!(x::AbstractVector; signs=ones(Bool, size(x)), scaling = 1)
+    ln = size(x,1)
+    ls = size(signs,1)
+    @assert rem(log(ln) / log(2), 1) ≈ 0 "Size of vector must be power of 2."
```

Comment:
Is this not the check you claim to avoid in the docstring?

Reply:
Yeah I saw this and removed the comment.